test/metabase/lib/test_util/generators/filters.cljc

```diff
@@ -0,0 +1,278 @@
+(ns metabase.lib.test-util.generators.filters
+  (:require
+   [metabase.lib.core :as lib]
+   [metabase.lib.options :as lib.options]
+   [metabase.lib.test-util.generators.util :as gen.u]
+   [metabase.lib.types.isa :as lib.types.isa]
+   [metabase.util.time :as u.time]))
+
+;; Filter values =================================================================================
+(defn- gen-int []
+  (- (rand-int 2000000) 1000000))
+
+(def ^:private valid-ascii
+  (mapv char (range 0x20 0x7f)))
+
+(def ^:private han-unicode
+  (mapv char (range 0x4e00 0xa000)))
+
+(defn- gen-string
+  ([]
+   (if (< (rand) 0.1)
+     (gen-string han-unicode 40)
+     (gen-string valid-ascii 70)))
+  ([symbols max-len]
+   (apply str (repeatedly (inc (rand-int max-len)) #(rand-nth symbols)))))
+
+(defn- gen-time []
+  (u.time/local-time (rand-int 24) (rand-int 60) (rand-int 60) (rand-int 1000000000)))
+
+(defn- gen-time:minute []
+  (u.time/local-time (rand-int 24) (rand-int 60) 0 0))
+
+(defn- gen-date []
+  ;; Random day of the year, from 2000-01-01 through 2037-12-31.
+  ;; Avoids 2038 because of the 32-bit timestamp overflow.
+  ;; TODO: Always adds 0-364 days, so it can't return Dec 31st of a leap year. I don't think it matters, but I will
+  ;; highlight it.
+  (u.time/add (u.time/local-date (+ 2000 (rand-int 38)) 1 1) ;; Jan 1, from 2000 through 2037
+              :day
+              (rand-int 365)))
+
+(defn- gen-datetime []
+  (u.time/local-date-time (gen-date) (gen-time)))
+
+(defn- gen-datetime:minute []
+  (u.time/local-date-time (gen-date) (gen-time:minute)))
+
+(defn- gen-latitude []
+  ;; +/- 75 degrees is a generous but plausible range for latitudes.
+  (* 150 (- (rand) 0.5)))
+
+(defn- gen-longitude []
+  ;; +/- 180 degrees
+  (- (* 360 (rand))
+     180))
+
+(def ^:private fake-categories
+  (vec (for [i (range 1 41)]
+         (str "Fake Category " i))))
+
+(defn- gen-category []
+  ;; Just some made-up values that are clearly not random strings for debugging.
+  (rand-nth fake-categories))
+
+(defn- rand-column-value [{:keys [effective-type] :as column}]
+  (cond
+    ;; Numeric PKs and FKs are always integers.
+    (and (lib.types.isa/id? column)
+         (lib.types.isa/numeric? column))              (abs (gen-int))
+    (#{:type/BigInteger :type/Integer} effective-type) (gen-int)
+    (lib.types.isa/category? column)                   (gen-category)
+    (lib.types.isa/latitude? column)                   (gen-latitude)
+    (lib.types.isa/longitude? column)                  (gen-longitude)
+    (lib.types.isa/numeric? column)                    (cond-> (gen-int)
+                                                         (< (rand) 0.5) (+ (rand)))
+    (lib.types.isa/string-or-string-like? column)      (gen-string)
+    (lib.types.isa/time? column)                       (gen-time)
+    (lib.types.isa/date-without-time? column)          (gen-date)
+    (isa? effective-type :type/DateTime)               (gen-datetime)
```

Comment:
nit: you could use `lib.types.isa/date-or-datetime?` to match how the FE does this check (for `:type/DateTime`), but I see that it could bring confusion

Reply:
Done. I'd rather be consistent and stick with the library functions, even if they're phrased backwards in some cases.